oidc.go

```diff
@@ -93,18 +94,23 @@ func NewProvider(ctx context.Context, issuer string) (*Provider, error) {
 	if err != nil {
 		return nil, err
 	}
+	defer resp.Body.Close()
+
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("unable to read response body: %v", err)
```

Comment:
- can this include the source url?
- most other errors from this package include an `oidc:` prefix, not sure if thats expected going forward, but should probably have it for consistency?